### PR TITLE
 Run End to End Tests before Release #271 

### PR DESCRIPTION
**Pull request by trulyronak**
This runs optic's end-to-end tests before any release, blocking release if they do not pass.

Before merging, note that will require a new org secret — a repo access token for end-to-end tests, called `END_TO_END_ACCESS_KEY`

After adding the `END_TO_END_ACCESS_KEY` token to the org secrets, this should be good to go

### CODE REVIEW

File: .github/workflows/endToEndTests.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+on:
+  pull_request:
+    branches:
+      - develop
+      - release
+  push:
+    branches:
+      - develop
+      - release
```
Comment (trulyronak) [inline .github/workflows/endToEndTests.yml:11]: this is one thing I'm not sure about — when should the end-to-end tests run? Right now I have it on any PR to `develop` or `release`, should I also be having it run once they merge into `develop` and `release` as well?

**Comment by trulyronak** [comment]
right now you can see that the tests fail since the `END_TO_END_ACCESS_KEY` doesn't yet exist - this should be set to be some repo_access_token that can read the optic_end_to_end_tests

**Comment by devdoshi** [comment]
I added the token, waiting for it to finish running now

File: .github/workflows/endToEndTests.yml (hunk 1 above)
Comment (devdoshi) [reply .github/workflows/endToEndTests.yml:11]: Seems fine for now, I'll be working on the e2e suite so I'll revisit once I look at it a bit more

**Pull request merged**